### PR TITLE
Tidy up package.json at dev-es6 branch #6775

### LINKED ISSUE #6775: Tidy up package.json at dev-es6 branch
For the lack of better issue title. Few things need to be done before moving forward with the ES6 conversion.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6775 

**Outline of Solution**

1. Resolved duplicate libraries in `package.json` used for building and `package.json` used for CDN.
1. Added new ESLint ruleset for ES6 files; based on Airbnb's base config. Commit 86a4685 of #6766 can pass this lint with very minor modification.
1. Updated babel libs to the latest.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
Passed the linting step; should be good to review.

File: package.json [hunk 1]
```diff
@@ -38,7 +21,7 @@
     "build:test": "babel src/main/webapp/dev/ -x \".es6\" -d src/main/webapp/dev/",
     "build:src": "babel src/main/webapp/js/ -x \".es6\" -d src/main/webapp/js/",
     "build": "npm run build:src && npm run build:test",
-    "lint:src": "eslint src/main/webapp/js -c static-analysis/teammates-eslint.yml --ext es6",
+    "lint:src": "eslint src/main/webapp/js/*.es6 -c static-analysis/teammates-eslint-es6.yml",
```
Comment (li-kai) [inline package.json]: this wouldn't work for nested `.es6` files

File: package.json [hunk 2]
```diff
@@ -1,29 +1,12 @@
 {
-  "dependencies": {
```
Comment (li-kai) [inline package.json]: keeping dependencies here is nice because we can do

`import 'bootbox' from 'bootbox'` in functions like `BootboxWrapper`

**Comment by wkurniawan07** [comment]
I just thought, rather than disabling `no-undef` and `no-use-before-define` globally, it's better to disable on individual files and fix file by file. Global disable needs global fix all at once.

**Pull request merged**